packages/sidecar/src/main.ts

```diff
@@ -91,7 +91,7 @@ const enableCORS = (handler: RequestHandler): RequestHandler =>
         res.end();
         return;
       }
-      return handler(req, res, pathname, searchParams);
+      return (await handler)(req, res, pathname, searchParams);
```

Comment:
I do not like `await`ing a handler every single time we get a request. It adds at least one tick to everything. Can we find a better way to do this? I'm happy doing `await mcp.connect()` at the start of the server and delay that once rather than doing this.